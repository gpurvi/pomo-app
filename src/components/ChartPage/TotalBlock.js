import React from 'react';
import TotalTable from "./TotalTable";
import {getTotal} from "../common/apiCalls";

export default class TotalBlock extends React.Component {
    constructor(props) {
        super(props);
        this.state = {
            totalSessions: 0,
            totalHours: 0,
            averageSessions: 0,
            averageHours: 0,
            error: ''
        };
    }

    async componentDidMount() {
        try {
            const total = await getTotal();
            this.setState(() => ({...total}));
        } catch (err) {
            this.setState(() => ({error: err.message}));
        }
    }

    render() {
        return (
            <TotalTable {...this.state} />
        );
    }
}
