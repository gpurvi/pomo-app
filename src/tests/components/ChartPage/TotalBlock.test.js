import React from 'react';
import {shallow} from 'enzyme';
import TotalBlock from '../../../components/ChartPage/TotalBlock';

it('should render ChartButtons', () => {
    const wrapper = shallow(<TotalBlock/>);
    expect(wrapper).toMatchSnapshot();
});
