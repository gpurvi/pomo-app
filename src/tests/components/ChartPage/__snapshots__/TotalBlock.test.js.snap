// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render ChartButtons 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <TotalBlock />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "averageHours": 0,
      "averageSessions": 0,
      "error": "",
      "totalHours": 0,
      "totalSessions": 0,
    },
    "ref": null,
    "rendered": null,
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "averageHours": 0,
        "averageSessions": 0,
        "error": "",
        "totalHours": 0,
        "totalSessions": 0,
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
